### PR TITLE
Add a Python Jenkins k8s agent image

### LINKED ISSUE #27: Add a Python Jenkins k8s agent image

### PR DESCRIPTION
**Pull request by yuezhuangshi**
Follow the existing specifications and add python3.7 jenkins k8s agent image

fix #27

### CODE REVIEW

**Review by LinuxSuRen — COMMENTED** [review]
Thanks for your contribution. We build this project in the GitHub actions. So, would you mind adding the build script of your PR into the GitHub actions?

By the way, I'm not familiar with Python programming. It would be great if you can provide a sample Python project. Then I can test this PR with it.

File: .github/workflows/build-podman.yaml [hunk 1]
```diff
@@ -290,3 +290,73 @@ jobs:
...
+      - name: Build and push Docker images for Contributors
+        uses: docker/build-push-action@v2.4.0
+        if: github.repository_owner != 'kubesphere'
+        with:
+          context: python
+          file: python/podman/Dockerfile
+          tags: ${{ steps.metaContributors.outputs.tags }}
+          push: ${{ github.event_name != 'pull_request' }}
+          labels: ${{ steps.metaContributors.outputs.labels }}
+          platforms: linux/amd64
```
Comment (LinuxSuRen) [inline .github/workflows/build-podman.yaml:362]: Please keep an empty line at the end of each file. See also [here](https://www.google.com.hk/search?q=why+add+a+new+line+at+the+end+of+this+file&oq=&sourceid=chrome&ie=UTF-8).

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: .github/workflows/build-podman.yaml (hunk 1 above)
Comment (yuezhuangshi) [reply .github/workflows/build-podman.yaml]: Done. 
BTW, We can use [EditorConfig](https://editorconfig.org/) to guarantee that behavior.

Comment (LinuxSuRen) [reply .github/workflows/build-podman.yaml]: Good idea.

**Comment by yuezhuangshi** [comment]
> Thanks for your contribution. We build this project in the GitHub actions. So, would you mind adding the build script of your PR into the GitHub actions?
> 
> By the way, I'm not familiar with Python programming. It would be great if you can provide a sample Python project. Then I can test this PR with it.

I have already added the build script, waiting for Github actions to build image.
Currently, I will build the image locally, and then use the project on github for testing

**Comment by yuezhuangshi** [comment]
@LinuxSuRen I have tested this image locally, sample project is [thefuck](https://github.com/nvbn/thefuck), sample workflow ref [here](https://github.com/nvbn/thefuck/blob/master/.github/workflows/test.yml) and the jenkinsfile I have been used are as follow:
```
pipeline {
    agent {
        label 'python37'
    }

    stages {
        stage('Checkout') {
            steps {
                git branch: 'master', credentialsId: 'github', url: 'git@github.com:nvbn/thefuck.git'
            }
        }
        stage('Install dependencies') {
            steps {
                container('python') {
                    sh '''
                      pip install -i https://pypi.tuna.tsinghua.edu.cn/simple -Ur requirements.txt coveralls
                      python setup.py develop
                    '''
                }
            }
        }
        stage('Run tests') {
            steps {
                container('python') {
                    sh 'coverage run --source=thefuck,tests -m pytest -v --capture=sys tests'
                }
            }
        }
    }
}
```

**Review by LinuxSuRen — APPROVED** [review]
I've tested these new images (see them below). They work well.

```
surenpi/builder-python:dev-v3.1.0-podman

surenpi/builder-python:dev-v3.1.0
```

**Comment by JohnNiang** [comment]
Hi @yuezhuangshi , thanks for your contriubution. 

I think we are missing docker image declaration in `README.md`. Can you help us add it in another PR?

**Comment by yuezhuangshi** [comment]
> Hi @yuezhuangshi , thanks for your contriubution.
> 
> I think we are missing docker image declaration in `README.md`. Can you help us add it in another PR?

Sure

**Pull request merged**